WalletWasabi.Fluent/ViewModels/MainViewModel.cs

```diff
@@ -190,12 +188,10 @@ private void RegisterViewModels()
 					return null;
 				});
 
-			LegalDocumentsViewModel.RegisterAsyncLazy(
-				async () =>
+			LegalDocumentsViewModel.RegisterLazy(
+				() =>
 				{
-					var content = await File.ReadAllTextAsync(_global.LegalDocuments.FilePath);
-
-					var legalDocs = new LegalDocumentsViewModel(content);
+					var legalDocs = new LegalDocumentsViewModel(_global.LegalChecker.CurrentLegalDocument?.Content ?? "");
```

Comment:
IMO Instead of showing empty content, we should show the progress ring in the LegalDocumentsView while the legal doc is downloading, once it is finished, hide the progress ring and show the content.

Reply:
ACK

Reply:
UITeam will have a small animation job here, added a TODO section.